storage/remote/write_handler.go

```diff
@@ -611,18 +623,18 @@ func (rw *rwExporter) ConsumeMetrics(ctx context.Context, md pmetric.Metrics) er
 		LookbackDelta:                     rw.lookbackDelta,
 		EnableTypeAndUnitLabels:           rw.enableTypeAndUnitLabels,
 	})
-	if err != nil {
-		rw.logger.Warn("Error translating OTLP metrics to Prometheus write request", "err", err)
-	}
+
+	defer func() {
+		if err != nil {
+			_ = app.Rollback()
```

Comment:
[Nit] Why not propagate the rollback error?
```suggestion
			if rollbackErr := app.Rollback(); rollbackErr != nil {
				err = errors.Join(err, rollbackErr)
			}
```

Reply:
Yeah I'm also not sure - I figured that it might be useful to know that the rollback failed, which isn't exactly ideal, right. You can see in `writeV2` though that the rollback error isn't ignored, but logged:
```golang
if rerr := app.Rollback(); rerr != nil {
	h.logger.Error("writev2 rollback failed on retry-able error", "err", rerr)
}
```